frontend/pages/labels/components/LabelForm/LabelForm.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { ReactNode, useState } from "react";
+
+import validate_presence from "components/forms/validators/validate_presence";
+
+// @ts-ignore
+import InputField from "components/forms/fields/InputField";
+import Button from "components/buttons/Button";
+
+export interface ILabelFormData {
+  name: string;
+  description: string;
+}
+
+interface ILabelFormProps {
+  defaultName?: string;
+  defaultDescription?: string;
+  additionalFields?: ReactNode;
+  isUpdatingLabel?: boolean;
+  onCancel: () => void;
+  onSave: (formData: ILabelFormData, isValid: boolean) => void;
+}
+
+const baseClass = "label-form";
+
+const LabelForm = ({
+  defaultName = "",
+  defaultDescription = "",
+  additionalFields,
+  isUpdatingLabel,
+  onCancel,
+  onSave,
+}: ILabelFormProps) => {
+  const [name, setName] = useState(defaultName);
+  const [description, setDescription] = useState(defaultDescription);
+  const [nameError, setNameError] = useState<string | null>("");
+
+  const onNameChange = (value: string) => {
+    setName(value);
+    setNameError(null);
+  };
+
+  const onDescriptionChange = (value: string) => {
+    setDescription(value);
+  };
+
+  const onSubmitForm = (evt: React.FormEvent) => {
+    evt.preventDefault();
+
+    let isFormValid = true;
```

Comment:
@jacobshandling @ghernandez345 

Frontend sync conversation: When should we valididate if a required field is empty. Ex: Here it's not until you try to submit the form, other forms don't allow you to click the save button until it's filled out, other's it shows the error that it's error until it's filled.